src/Makefile

```diff
@@ -153,7 +153,12 @@ all_platforms:
 
 command.c: include/version.h
 
+VERSION_STRING := $(shell echo "#define FIRMWARE_VERSION \"`git describe --always --dirty --tags`\"")
+
 include/version.h: FORCE
-	$(Q)echo " GIT include/version.h"
-	$(Q)echo "#define FIRMWARE_VERSION \"$(shell git describe --always --dirty --tags)\"" > $@
+ifneq ($(file <include/version.h), $(VERSION_STRING))
```

Comment:
This introduces what looks like it might be a bash-ism, it might be worth having the rule be `$(Q)./gen-version.sh` or so, and have a script based on /bin/sh to regen the header as-needed

Reply:
not sure a separate script pays off here, i hope i've simplified the code now so it works with other shells too :)

Reply:
The way you fixed it is also valid (hopefully won't cause issues with older GNU Make and make-alternatives). The reason we suggested a standalone script is so you wouldn't tangle with other bash-alternative shells, not because it was the only workable option :)